chicago-fullnote-bibliography-fr.csl

```diff
@@ -594,34 +678,62 @@
             <text term="edition"/>
           </group>
           <group delimiter=" ">
-            <text term="section"/>
+            <text term="section" form="short"/>
             <text variable="section"/>
           </group>
-          <choose>
-            <if variable="locator" match="none">
-              <text variable="page"/>
-            </if>
-            <else>
-              <group delimiter=" ">
-                <choose>
-                  <if locator="page" match="none">
-                    <label variable="locator" form="short" suffix=" "/>
-                  </if>
-                </choose>
-                <text variable="locator"/>
-              </group>
-            </else>
-          </choose>
         </group>
       </if>
     </choose>
   </macro>
+  <macro name="event-note">
+    <text variable="event"/>
+  </macro>
   <macro name="event">
-    <group delimiter=" ">
-      <text term="presented at"/>
-      <text variable="event"/>
+    <choose>
+      <if variable="title">
+        <group delimiter=" ">
+          <text term="presented at"/>
+          <text variable="event"/>
+        </group>
+      </if>
+      <else>
+        <group delimiter=" ">
+          <text term="presented at" text-case="capitalize-first"/>
+          <text variable="event"/>
+        </group>
+      </else>
+    </choose>
+  </macro>
+  <macro name="originally-published">
+    <group delimiter=", ">
+      <group delimiter="&amp;#160;: ">
```

Comment:
I'm pretty sure you want `&#160;` here and in all other instances where you use it. This is true both for this and the author-date style